enhancements/security/manual-cloud-credentials.md

```diff
@@ -0,0 +1,170 @@
+  * Users can upgrade without failures when the new release image contains credential changes.
+  * Users have tooling at their disposal to aid them in detecting what credentials changes have occurred in a release image.
+  * Users have tooling at their disposal to aid them in updating their credentials with new permissions.
+
+### Non-Goals
+
+What is out of scope for this proposal? Listing non-goals helps to focus discussion
+and make progress.
+
+  * Providing a mechanism to disable the cloud credential operator, this mechanism already exists.
+
+## Proposal
+
+### Implementation Details/Notes/Constraints [optional]
+
+This proposal requires the CCO to begin reporting Upgradable=False in a number of scenarios, unless an admin informs us they have taken action to reconcile new permissions and it is safe to upgrade.
+
+CLI tooling will be implemented to help users who wish to disable the CCO to perform their initial installations, update credentials prior to upgrade for release images available for update, and reconcile credentials against current CredentialsRequests in the cluster on an on-going basis. Users must take this manual step to clear the Upgradable=False condition if credentials have changed in any available update.
+
+Cloud credential operator reconcile code will be refactored to be a little more friendly to use as a vendored library, for use in the CLI.
+
+Note that when we say CCO is disabled, it is still running, just not minting credentials, and does not have access to the powerful "root" credentials for the cloud provider account.
+
+### Risks and Mitigations
+
+
+## Design Details
+
+Core of this proposal will be CLI tooling which vendors code from the credentials operator and allows for single use execution of the reconcile loop. Provising CLI tooling allows users to bypass the security concerns of storing powerful cloud credentials in the cluster and instead keep them entirely locally.
+
+### Upgradable False Condition
+
+A field will be added to the cloud credential operator ConfigMap (possibly becoming a separate API type in another enhancement) storing a list of Updates (from ClusterVersion.AvailableUpdates) that have been reconciled, confirming that we have minted credentials and it is safe to upgrade to this release.
+
+```yaml
+  reconciledAvailableUpdates:
+  - version: 4.4.1
+    image: quay.io/openshift-release-dev/ocp-release:4.4.1-x86_64
+  - version: 4.4.2
+    image: quay.io/openshift-release-dev/ocp-release:4.4.2-x86_64
+```
+
+Entries will be added to this list by the CLI tooling below, however an administrator could do this by hand if desired.
+
+If the CCO is enabled per normal best practice install, this field will be ignored, and should be absent.
+
+However if CCO is disabled, it will compare the ClusterVersion.Spec.AvailableUpdates to this list of ReconciledAvailableUpdates. If *any* available update is not present in the reconciled list, the CCO will report Upgradable=False.
+
+### Initial Installation
```

Comment:
1. What is the source of CredentialsRequests objects.
2. What inputs will the cli require to update CredentialsRequests and create secret objects.
3. What about the source or type of creds used to interact with cloud APIs
4. What about various invariants wrt access cloud APIs like endpoints, trust etc.
5. Create CredentialsRequests with new information during bootstrap will race with CVO trying to create those.
6. I don't want to use cred-minter cli to create these credentials, I have my own way of doing those. How do i get what's needed and create them on my own.

Reply:
1. Release image as before.
2. As far as I can see, it should just be the AWS cred env vars being set (or a local creds file, to match whatever installer does), and the path to the manifests dir from openshift-install.
3. Not sure what you  meant with this one.
4. For endpoints and trust how are those configured at install time today, InstallConfig?
5. No CredentialsRequests should be created with any new information during bootstrap, I would expect only new Secrets written to disk as manifests prior to create-cluster, and if we are doing some hack to get *status* onto the CredentialsRequest it would be an annotation added to the CredentialsRequest manifest, but again prior to create-cluster, nothing should be changing during bootstrap as far as I'm aware. 
6. Should be same as today, create manifests, edit to disable CCO, drop in your secret yamls, and then create-cluster. (With addition of a command to parse out cred requests from a release image and display the permissions perhaps, I will add that to proposal, it should be easy to just view the cred requests)